public/components/freelancer-edit.js

```diff
@@ -0,0 +1,427 @@
+          <span>
+            <input type="checkbox" name="recurrence-days" ref="recurrence-days" id="emergency-form-recurrence-day" onClick={this.updateCheck} onChange={this.onCheckChange} value="4" />
+            <label>Th</label>
+            From <input type="time" id="emergency-time-4-start" disabled /> to <input type="time" id="emergency-time-4-end" disabled />
+            <input type="text" id="emergency-location-4" placeholder="Location" disabled />
+          </span>
+          <span>
+            <input type="checkbox" name="recurrence-days" ref="recurrence-days" id="emergency-form-recurrence-day" onClick={this.updateCheck} onChange={this.onCheckChange} value="5" />
+            <label>Fr</label>
+            From <input type="time" id="emergency-time-5-start" disabled /> to <input type="time" id="emergency-time-5-end" disabled />
+            <input type="text" id="emergency-location-5" placeholder="Location" disabled />
+          </span>
+          <span>
+            <input type="checkbox" name="recurrence-days" ref="recurrence-days" id="emergency-form-recurrence-day" onClick={this.updateCheck} onChange={this.onCheckChange} value="6" />
+            <label>Sa</label>
+            From <input type="time" id="emergency-time-6-start" disabled /> to <input type="time" id="emergency-time-6-end" disabled />
+            <input type="text" id="emergency-location-6" placeholder="Location" disabled />
+          </span>
+          <span>
+            <input type="checkbox" name="recurrence-days" ref="recurrence-days" id="emergency-form-recurrence-day" onClick={this.updateCheck} onChange={this.onCheckChange} value="0" />
+            <label>Su</label>
+            From <input type="time" id="emergency-time-0-start" disabled /> to <input type="time" id="emergency-time-0-end" disabled />
+            <input type="text" id="emergency-location-0" placeholder="Location" disabled />
+          </span>
+        </div>
+
+        <div id="emergency-form-repetition-type">
+          <label>Use schedule from:
+            <input type="date" id="emergency-repetition-start-date" defaultValue={new Date().toJSON().slice(0,10)}/>
+          </label>
+          <label><input type="radio" name="emergency-repetition-type" value="weeks" defaultChecked={true} onChange={this.onRadioChange} />
+            <input type="text" id="emergency-repetition-weeks" placeholder="For a number of weeks" required/>
+          </label>
+          <label>
+            <input type="radio" name="emergency-repetition-type" value="until" onChange={this.onRadioChange} />
+            Until <input type="date" id="emergency-repetition-end-date" defaultValue={new Date().toJSON().slice(0,10)} disabled/>
+          </label>
+        </div>
+
+        <input type="submit" value="Update weekly schedule"/>
+      </form>
+    );
+  }
+}
+
+class FreelancerEmergencyForm extends React.Component {
+  render() {
+    return (
+      <div>
+        <FreelancerSingleDateForm />
```

Comment:
Would invert the order, first the schedule, then the single days. So it's also more intuitive I'd say.